drivers/gles2/rasterizer_storage_gles2.cpp

```diff
@@ -5612,6 +5612,15 @@ void RasterizerStorageGLES2::initialize() {
 	config.pvrtc_supported = config.extensions.has("IMG_texture_compression_pvrtc");
 	config.support_npot_repeat_mipmap = config.extensions.has("GL_OES_texture_npot");
 
+	// If the desktop build is using S3TC, and you export / run from the IDE for android, if the device supports
+	// S3TC it will crash trying to load these textures, as they are not exported in the APK. This is a simple way
+	// to prevent Android devices trying to load S3TC, by faking lack of hardware support.
+#ifndef TOOLS_ENABLED
+#if defined ANDROID_ENABLED
```

Comment:
Sorry I've been away from internet (and will be until at least tomorrow afternoon I expect). Yup, I did notice the potential for the same issue to occur on iOS and javascript, although I can't confirm or not whether those do occur in practice (as iOS is quite fixed hardware, and I haven't been able to get an HTML5 export working for quite some time!).

The quick fix here does work for this specific case, and according to some hardware stats I looked at, it might affect up to 1/5 of android devices (those that support S3TC, and they would probably also crash with this). It is possible to workaround by unticking S3TC in the compressed texures bit of project settings, and reimporting textures before making the final export for android. However this is easy to forget, and not at all obvious. So it is potentially quite an easy way of expanding the compatibility with Android devices for 3.2.

However I am in agreement that it is a bit hacky and is potentially the symptom of a larger problem whereby the features in the export are not taken into account at runtime, it's something that we (or probably me lol) need to ask reduz (is it likely to be a one off bug, or a systemic problem) either in PR meeting or on IRC.

Reply:
For Javascript it's more complex than that, as it does use S3TC on desktop browsers, but likely something else on mobile browsers (didn't check what exactly).

Reply:
Rather than modifying the import files, it might be easier to use the approach of saving which features have successfully been exported (maybe this is done already? A long time since I looked at this), loading this feature set in at runtime and perhaps converting to bitflags, then checking the bitflags before attempting to load a resource. This would solve the problem in a generic way as it could potentially occur in several areas.

Or we could simply log a warning message when failing to load a resource, and go for the next one in the list, rather than crashing... :smile: 

I'll try and have another look at this problem after the weekend.